### ISSUE TITLE
The tempbadmols.txt is empty after python text_sample.py

### ISSUE THREAD

**Issue by yxjacksonyyds** [CRIPAC-DIG/tgm-dlm#9]
你好，十分感谢您的分享，我在复现代码时候，遇到了一些问题，希望得到您的帮助：
我在训练好检查点后，执行python text_sample.py命令后，在项目根目录下生成了tempbadmols.txt文件，用于post_sample.py，但是生成的该txt文件是空值。在post_sample代码中报错。
以下是我调试的结果：
将text_sample的部分代码后面加入调试信息：
    sample = th.concat(allsample,dim=0)
    print('decoding for e2e', )
    print(sample.shape)
    x_t = th.tensor(sample).cuda()
    # x_t = sample.clone().detach().cuda()
    reshaped_x_t = x_t
    logits = model.get_logits(reshaped_x_t)  # bsz, seqlen, vocab
    cands = th.topk(logits, k=1, dim=-1)
    

> print(f"cands在这里啊：{cands}")

    sample = cands.indices
    sample = sample.squeeze(-1)

>     print(sample)

    from mytokenizers import regexTokenizer
    tokenizer = regexTokenizer()
    c = tokenizer.decode(sample)
**以下是输出的内容：**
cands在这里啊：torch.return_types.topk(
values=tensor([[[11.3962],
         [12.0005],
         [10.8855],
         ...,

         …,
         [13.6369],
         [14.3101],
         [14.1626]]], device='cuda:0', grad_fn=<TopkBackward0>),
indices=tensor([[[0],
         [0],
         [0],
         …,
         [0],
         [0],
         [0]]], device='cuda:0'))
tensor([[0, 0, 0,  ..., 0, 0, 0],
        [0, 0, 0,  ..., 0, 0, 0],
        [0, 0, 0,  ..., 0, 0, 0],
        ...,
        [0, 0, 0,  ..., 0, 0, 0],
        [0, 0, 0,  ..., 0, 0, 0],
        [0, 0, 0,  ..., 0, 0, 0]], device='cuda:0')。
明显看出，在cand.indice阶段出现了问题导致在取最大可能值的索引时，错误的取到了全0，进而导致在解码后填充【PAD】，导致输出全空。
请问，是哪里出问题了呢？是否是环境问题或者参数问题？（我在跑train的时候没有跑200000条数据，仅仅只跑了2000条数据，会和这个有关系吗？
）

**Comment by Deno-V** [comment]
是的，因为采用的diffusion框架在现在（2025年）看起来是有点旧了，训练的效率并不够，现在您可以尝试更新的框架，例如flow matching等等，在这一套老方案下，2000steps是完全不够用的，不到10w步应该是生成不了什么像样的结果（正如传统的图像的diffusion model）
